packages/paste-core/components/file-picker/src/index.tsx

```diff
@@ -0,0 +1,78 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {Label, RequiredDot} from '@twilio-paste/label';
+import {Text} from '@twilio-paste/text';
+import {useUID} from '@twilio-paste/uid-library';
+import {Button} from '@twilio-paste/button';
+import {MediaObject, MediaBody, MediaFigure} from '@twilio-paste/media-object';
+import {Box} from '@twilio-paste/box';
+import type {BoxProps} from '@twilio-paste/box';
+
+export interface FilePickerProps extends React.HTMLAttributes<HTMLInputElement>, Pick<BoxProps, 'element'> {
+  disabled?: boolean;
+  required?: boolean;
+}
+
+const FilePicker = React.forwardRef<HTMLInputElement, FilePickerProps>(
+  ({element = 'FILEPICKER_CONTAINER', disabled = false, required = false, ...props}, ref) => {
+    const [fileName, setFileName] = React.useState('No files selected');
+
+    const id = useUID();
+
+    const handleChange = (evt: any): void => {
+      const file = evt.target.value;
+      setFileName(file);
+      console.log('the event', evt);
+      console.log('file name', fileName);
+    };
+
+    return (
+      <MediaObject {...props}>
```

Comment:
Suggestion: as the component is a react replacement, for the native file input element, I suspect the intended target for additional props should be the same element that the ref goes on it. Safely spread, of course.